src/common/Statistics.h

```diff
@@ -0,0 +1,71 @@
+/********************************************************************
+ * 2023 -
+ * open source under Apache License Version 2.0
+ ********************************************************************/
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIBHDFS3_STATISTICS_H
+#define LIBHDFS3_STATISTICS_H
+
+#include <cstdint>
+#include <chrono>
+#include <mutex>
+#include <thread>
+
+namespace Hdfs {
+namespace Internal {
+    enum ReadLen {
+        LESS_16K,
+        LESS_128K,
+        LESS_1M,
+        LESS_6M,
+        BIG_6M,
+        TYPE_COUNT
+    };
+
+    class Statistics {
+    public:
+        Statistics();
+        ~Statistics();
+        void addReplicaReadStatistic(int64_t len, int64_t cost);
+        void addEcReadStatistic(int64_t len, int64_t cost);
+
+    private:
+        void addReadStatistic(int64_t (&read_count)[TYPE_COUNT],
+                              int64_t (&read_cost)[TYPE_COUNT],
+                              int64_t len,
+                              int64_t cost);
+        void printStatistic();
+
+        void printWork();
+
+    private:
+        int64_t replica_read_count[TYPE_COUNT];
+        int64_t replica_read_cost[TYPE_COUNT];
+        int64_t ec_read_count[TYPE_COUNT];
+        int64_t ec_read_cost[TYPE_COUNT];
+        std::thread printThread;
+        bool running;
+        bool enablePrint;
+        std::mutex mtx;
```

Comment:
what is the purpose of the mutex? `addReadStatistic` doesn't update any field of this class

Reply:
Another thread will print the statistics arrays every 60 seconds and clears it.